server/api.py

```
@@ -61,25 +78,31 @@ async def process_pending_signals():
                 storage.mark_signal_processed(signal.signal_id)
-                
                 print(f"[OK] Alert {alert.alert_id} created from signal {signal.signal_id}")
-        
+
+            except Exception as e:
+                print(f"[WARN] Error processing signal: {e}")
+            finally:
+                storage.pipeline_running = False
+                storage.last_pipeline_run = time.time()
+
         except Exception as e:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)